### PR TITLE
Example of auto notification tracking

### PR DESCRIPTION
**Pull request by bsneed**
- Added example of automatic notification tracking for iOS/tvOS/watchOS.
- Fixed incorrect method signatures in iOS lifecycle plugin.

### CODE REVIEW

File: Examples/apps/MacExample/MacExample/AppDelegate.swift [hunk 1]
```diff
@@ -22,7 +22,7 @@ class AppDelegate: NSObject, NSApplicationDelegate {
             .flushInterval(10)
         
         analytics = Analytics(configuration: configuration)
-
```
Comment (migs647) [inline Examples/apps/MacExample/MacExample/AppDelegate.swift:25]: Should we maybe add a comment: `// Uncomment to add notification tracking to your app` or something?

File: Sources/Segment/Plugins/Platforms/iOS/iOSLifecycleMonitor.swift [hunk 2]
```diff
@@ -10,6 +10,8 @@
 import Foundation
 import UIKit
 
+// NOTE: These method signatures are marked optional as application extensions may not have
```
Comment (migs647) [inline Sources/Segment/Plugins/Platforms/iOS/iOSLifecycleMonitor.swift:13]: 👍🏻

**Review by migs647 — APPROVED** [review]
LGTM

File: Examples/apps/MacExample/MacExample/AppDelegate.swift (hunk 1 above)
Comment (bsneed) [reply Examples/apps/MacExample/MacExample/AppDelegate.swift:25]: It's a placeholder until I get full lifecycle stuff on Mac working.

**Pull request merged**